server/src/main/java/org/elasticsearch/health/node/DiskHealthInfo.java

```diff
@@ -8,20 +8,45 @@
 
 package org.elasticsearch.health.node;
 
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
 import org.elasticsearch.health.HealthStatus;
 
+import java.io.IOException;
+
 /**
  * The health status of the disk space of this node along with the cause.
  */
-record DiskHealthInfo(HealthStatus healthStatus, Cause cause) {
+public record DiskHealthInfo(HealthStatus healthStatus, Cause cause) implements Writeable {
     DiskHealthInfo(HealthStatus healthStatus) {
         this(healthStatus, null);
     }
 
-    enum Cause {
-        NODE_OVER_HIGH_THRESHOLD,
-        NODE_OVER_THE_FLOOD_STAGE_THRESHOLD,
-        FROZEN_NODE_OVER_FLOOD_STAGE_THRESHOLD,
-        NODE_HAS_NO_DISK_STATS
+    public DiskHealthInfo(StreamInput in) throws IOException {
+        this(in.readEnum(HealthStatus.class), in.readOptionalEnum(Cause.class));
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        healthStatus.writeTo(out);
+        out.writeOptionalEnum(cause);
+    }
+
+    public enum Cause {
+        NODE_OVER_HIGH_THRESHOLD((byte) 0),
+        NODE_OVER_THE_FLOOD_STAGE_THRESHOLD((byte) 1),
+        FROZEN_NODE_OVER_FLOOD_STAGE_THRESHOLD((byte) 2),
+        NODE_HAS_NO_DISK_STATS((byte) 3);
+
+        private final byte value;
```

Comment:
Hm, in the beginning I thought adding specific values and implementing our own deserialization would make it less fragile to break during a maintenance. Somebody wouldn't accidentally change the order or something and break backwards compatibility. However, since we are using the built-in writable implementation, the `value` is not used at all anymore. I will remove it.